api/v1alpha1/mustgather_types.go

```diff
@@ -55,12 +47,55 @@ type MustGatherSpec struct {
 	// +kubebuilder:validation:Format=duration
 	MustGatherTimeout metav1.Duration `json:"mustGatherTimeout,omitempty"`
 
+	// The target location for the must-gather bundle to be uploaded to.
+	// If not specified, the bundle will not be uploaded.
+	// +kubebuilder:validation:Optional
+	UploadTarget *UploadTargetSpec `json:"uploadTarget,omitempty"`
+}
+
+// SFTPSpec defines the desired state of SFTPSpec
+// +kubebuilder:validation:XValidation:rule="size(self.caseID) > 0",message="caseID must not be empty"
+// +kubebuilder:validation:XValidation:rule="size(self.caseManagementAccountSecretRef.name) > 0",message="caseManagementAccountSecretRef.name must not be empty"
+type SFTPSpec struct {
```

Comment:
Looks like the host for the SFTP is out of scope here, are we considering that separately?